vortex-array/src/builders/primitive.rs

```diff
@@ -45,6 +46,55 @@ impl<T: NativePType> PrimitiveBuilder<T> {
             None => self.append_null(),
         }
     }
+
+    pub fn patch(&mut self, patches: Patches, starting_at: usize) -> VortexResult<()> {
+        let (array_len, offset, indices, values) = patches.into_parts();
+        assert!(starting_at + array_len == self.len());
+        let indices = indices.into_primitive()?;
+        let values = values.into_primitive()?;
+        let validity = values.validity_mask()?;
```

Comment:
if the validity is all true, then remove the check from the below loop.

Reply:
Done.